kbase-extension/static/kbase/js/common/jobCommChannel.js

```diff
@@ -205,19 +196,25 @@ define([
          */
         handleCommMessages(msg) {
             const msgType = msg.content.data.msg_type;
-            const msgData = msg.content.data.content;
-            const msgDataType = Utils.objectToString(msgData);
+            let msgData = msg.content.data.content;
 
-            // message data should be an object
-            if (msgDataType !== 'Object') {
-                return this.reportCommMessageError({ msgType, msgData });
+            // validate and discard any invalid messages
+            const validated = Jobs.validateMessage({ type: msgType, message: msgData });
+            // report any invalid data
+            if ('invalid' in validated) {
+                this.reportCommMessageError({ msgType, msgData: validated.invalid });
             }
+            // if there is no valid data, abort
+            if (!('valid' in validated)) {
```

Comment:
it's possible that `valid` could be `null`, so explicitly check that `validated.valid` is not defined